src/Microsoft.AspNet.Server.Kestrel/Infrastructure/MemoryPoolIterator2.cs

```diff
@@ -170,6 +174,11 @@ public int Seek(int char0)
                 }
                 while (block.End != index)
                 {
+                    if (_processed > _maxHeaderLength)
```

Comment:
I also think that the limits need to be passed as arguments to functions like Seek. Just in case it is ever used from something other than headers.